fairlearn/adversarial/_preprocessor.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+from ._constants import (
+    _TYPE_COMPLIANCE_ERROR,
+    _TYPE_CHECK_ERROR,
+    _ARG_ERROR_MESSAGE,
+    _TYPE_UNKNOWN_ERROR,
+)
+from sklearn.utils import check_array
+from sklearn.preprocessing import OneHotEncoder
+from pandas import Series, DataFrame
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.multiclass import type_of_target
+from numpy import all as np_all
+from numpy import sum as np_sum
+
+
+class FloatTransformer(BaseEstimator, TransformerMixin):
+    """
+    Transformer that maps dataframes to numpy arrays of floats.
```

Comment:
could you better explain here what it does and why it's needed?

Reply:
The description is making it sound like an sklearn pipeline

Reply:
I changed it a bit, is there anything I'm missing still perhaps?

Reply:
Reading the description, my question is then why not use a columntransformer?

Reply:
@adrinjalali my apologies I missed your response. A columntransformer such as in the example notebook would have worked fine too yes, and I had that at some point in time. But I decided I don't want to, because I don't want to deal with defining loss functions for mixed type columns. Say you pass two categorical columns for `y`. Then, how should we define a loss function for this? We have thought about multi-output loss functions, but decided it was too strange. Maybe a good option would have been to do the columntransform nonetheless, and then just raise an error if the user does not supply a loss function with it. That should be do-able besides some minor detail I won't mention. With more thought, there may be something better, but I think we should see if there is desire for this first.

Reply:
What I meant was to have a `ColumnTransformer` that does exactly what this transformer does. The output would be an array of floats, and then nothing would be different. You'd apply your ColumnTransformer internally on your `y`.